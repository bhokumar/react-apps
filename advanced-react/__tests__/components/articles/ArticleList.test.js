import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from '../../../lib/components/articles/ArticleList';

describe('ArticleList',() => {
  const testProps = {
    articles : {
      a: {id:'a'},
      b : {id : 'b'}
    },
    articleActions : {
      lookUpAuthor : jest.fn(() => ({}))
    }
  };

  it('ArticleList rnders correctly', () => {
    const tree = renderer.create(
      <ArticleList {...testProps}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(2);
  });
});
