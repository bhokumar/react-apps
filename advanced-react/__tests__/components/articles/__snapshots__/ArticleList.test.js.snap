// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList ArticleList rnders correctly 1`] = `
<div>
  <div>
    <div />
    <div />
    <div>
      <a />
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <a />
    </div>
    <div />
  </div>
</div>
`;
